perun-oidc-server/src/main/java/cz/muni/ics/oidc/server/claims/sources/Ga4ghApiClaimSource.java

```diff
@@ -0,0 +1,81 @@
+package cz.muni.ics.oidc.server.claims.sources;
+
+
+import com.fasterxml.jackson.databind.JsonNode;
+import cz.muni.ics.oidc.server.claims.ClaimSource;
+import cz.muni.ics.oidc.server.claims.ClaimSourceInitContext;
+import cz.muni.ics.oidc.server.claims.ClaimSourceProduceContext;
+import cz.muni.ics.oidc.server.claims.ClaimUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.Set;
+
+@Slf4j
+public class Ga4ghApiClaimSource extends ClaimSource {
+
+    private static final String ENDPOINT = "endpoint";
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
+    private static final String PARAM_NAME = "param_name";
+    private static final String EPPN = "{eppn}";
+
+    private final String endpoint;
+    private final String username;
+    private final String password;
+    private final String paramName;
+
+    public Ga4ghApiClaimSource(ClaimSourceInitContext ctx) {
+        super(ctx);
+
+        this.endpoint = ClaimUtils.fillStringMandatoryProperty(ENDPOINT, ctx, getClaimName());
+        this.username = ClaimUtils.fillStringMandatoryProperty(USERNAME, ctx, getClaimName());
+        this.password = ClaimUtils.fillStringMandatoryProperty(PASSWORD, ctx, getClaimName());
+
+        this.paramName = ClaimUtils.fillStringPropertyOrDefaultVal(PARAM_NAME, ctx, EPPN);
+
+        log.debug("{} - initialized", getClaimName());
```

Comment:
Maybe extend the log with more details of what properties have been used - this will make debugging in case of misconfiguration easier. Just leave out the password so it does not appear in the logs.